Initial version of the dotfiles.

smebberson opened the pull request:
This is a first commit (taken from smebberson/dotfiles) for an Idearium dotfiles. The intent is that all Idearium developers would use this dotfiles to improve developer productivity and to allow a tighter integratiob between Idearium development practices and the command line.

Please have a review of all code. You can also test it on your local to make sure it all works. Yell out if there any glaring omissions or issues you think should stop this from being the first version.

smebberson committed:
Initial version of the dotfiles (imported from smebberson/dotfiles).

allanchau commented on platform/linux/.gitconfig:
```diff
@@ -0,0 +1,9 @@
+[user]
+	name = Scott Mebberson
+	email = scott@scottmebberson.com
```
Hmm is there a way we can have this without specifying the user?

smebberson replied:
Oh right, indeed.

smebberson replied:
I'll have to improve this via the install process. I'll do that shortly.

allanchau commented on z.sh:
```diff
@@ -0,0 +1,243 @@
+# Copyright (c) 2009 rupa deadwyler under the WTFPL license
```
This has been updated to a v2 although it seems to work the same.

allanchau reviewed (commented):
Some discussion points, I'll add more in the comments.

allanchau commented:
I haven't used dot file on linux but we should add `.inputrc` on Mac https://github.com/allanchau/dotfiles/blob/master/src/.inputrc

What do you reckon about adding `.bash_prompt`? https://github.com/allanchau/dotfiles/blob/master/src/.bash_prompt
I have mine setup to show `master ✓ ~/Developer/dotfiles $ ` but it is personal preference I guess.

Aliases? https://github.com/allanchau/dotfiles/blob/master/src/.alias

Not everyone uses Homebrew, but do you know of a native way to install + setup apps? https://github.com/allanchau/dotfiles/blob/master/scripts/brew-cask.sh https://github.com/allanchau/dotfiles/blob/master/apps/atom/setup.sh

smebberson committed:
Updated the install process to allow customisation of git user.name and user.email configuration.

smebberson commented:
@allanchau, there were a bunch of missing files :( I've added them back in. Also update the `./install` process to ensure the user can customise the `git config` `user.name` and `user.email` values. Please take another look.

allanchau commented on .bash_aliases:
```diff
@@ -0,0 +1,42 @@
+alias vs='vagrant ssh'
+
+# git command shortcuts
+alias gs='git status'
```
`git status -s` is much prettier :)

smebberson replied:
Didn't know about that. Added :)

allanchau commented on .bash_aliases:
```diff
@@ -0,0 +1,42 @@
+
+# vagrant shortcuts
+alias v='vagrant'
+alias vs='vagrant ssh'
```
I also have these.
```shell
alias vh="vagrant halt"
alias vu="vagrant up"
```

smebberson replied:
Added

smebberson committed:
Added .inputrc

allanchau commented on .bash_aliases:
```diff
@@ -0,0 +1,42 @@
+
+function gbdo () {
+	git branch -D "$1" && git push origin ":$1"
+}
```
These also comes in handy for me, depends how lazy you feel lol
```shell
# dir traversal
alias ..="cd .."
alias ...="cd ../.."
alias ....="cd ../../.."

# Recursively remove Apple meta files
alias cleanupds="find . -type f -name '*.DS_Store' -ls -exec /bin/rm {} \;"
```

smebberson replied:
Added those.

allanchau replied:
@smebberson One more thing, can we add an alias for git? `g`

smebberson commented:
@allanchau, just added `.inputrc`.

smebberson committed:
Updated z.sh to version 2.

allanchau commented:
@smebberson I don't really want to override my existing setup, so do you think it is ok if I don't run the `install` script?

smebberson committed:
Updated the bash aliases.

allanchau commented on platform/linux/.gitconfig:
```diff
@@ -0,0 +1,6 @@
+[credential]
+	helper = cache --timeout=172800
+[diff]
+	external = /home/smebberson/diff.sh
```
@smebberson  This one still points to `smebberson`, I think the diff file is missing?

smebberson replied:
Thanks. Fixed that.

smebberson commented:
@allanchau, just added those aliases that you have.

smebberson committed:
Removed customisation of 'diff' in .gitconfig.

smebberson committed:
Updated Bash prompt based on Allan's.

smebberson commented:
@allanchau, I've update the `.bash_prompt` to be more like yours.

smebberson commented:
@allanchau, why don't you want to override your setup? Isn't that the point of having these? So that all Idearium devs run the same dotfiles and benefit from updates and improvements to them?

smebberson committed:
Removed another reference to smebberson.

allanchau commented:
@smebberson Yeah but I have some specific ones that wouldn't be useful for everyone `Homebrew`, `Logitech options` etc. Also I've set mine up to customise atom + Hyper with light themes which most people hate 👎 .

smebberson commented:
@allanchau, can we allow an avenue for customisation?

smebberson commented:
@allanchau, the reason I want this is that the devops environment depends on a few settings. So I was thinking this would be a good place to create them so that its super simple to get started as an Idearium developer.

allanchau commented:
@smebberson It would be good if the install script could run some custom code in the install script `custom.sh`? We can then move any customisations there? I'll run the install now

allanchau commented:
@smebberson is it possible to customise the dotfile path?

smebberson commented:
@allanchau, what do you mean by the dotfile path? What are you thinking through?

smebberson commented:
@allanchau, `custom.sh` is a good idea.

allanchau commented:
I currently have personal projects in `~/Developer/` and idearium projects in `~/Developer/idearium/`. It helps me scan over projects better but others may want it in `~/Library/Web` etc?

smebberson committed:
Added the ability to add customisations during the install process.

smebberson commented:
@allanchau, just added support for a `custom.sh`. Check it out and let me know if it does/doesn't suit your needs...

smebberson commented:
@allanchau, I think I can make it so that this repository can live anywhere. Let me have a crack at it.

smebberson commented:
@allanchau, is the `custom.sh` good enough?

smebberson committed:
Added some flexibility as to where the dotfiles repository can live.

smebberson commented:
@allanchau, this repository can now live anywhere and still work :)

allanchau commented:
@smebberson I am getting a few errors on install:
```shell
/Users/allanchau/.bash_profile: line 14: /Users/allanchau/.bash_colors: No such file or directory
/Users/allanchau/.bash_profile: line 14: /Users/allanchau/.bash_prompt: No such file or directory
/Users/allanchau/.bash_profile: line 14: /Users/allanchau/.bash_exports: No such file or directory
```

smebberson commented:
@allanchau, I just ran it on @alicecoleman's MacBook Pro with no issue at all. You said you did a search and replace on some paths, do you think that could be the issue? Can you please remove any changes you've made and try again?

smebberson commented:
@allanchau, I just did a search on the code base, and there is no reference to `Library`, other than the `README.md`. Do you have the latest version?

smebberson committed:
Updated the README.md with some more information.

allanchau commented:
@smebberson I think I found the problem, `.bash_exports` is missing? I added that and the install worked.

smebberson committed:
Seperated a general export from the mac specific file.

smebberson commented:
@allanchau, okay, I've just made a change that should update this to work better. Could you please try again?

allanchau commented:
@smebberson What do you reckon about replacing `dt` with the `dtl` alias? I've never actually used `git diff` on it's own.
Also some more aliases for you to consider 
```shell
alias gm='git mergetool' # instead of gmt?
alias gpl='git pull --rebase' # Add --rebase
```

smebberson committed:
Added a couple more aliases, and ordered them alphabetically by the full command.

smebberson commented:
@allanchau, I added `--rebase` to `gpl`. I also added `gm` and `gmt`. I think we need both. I use `gm` and `gmt` from time to time. Like wise with `gd` and `gdt`.

allanchau commented:
@smebberson Cool, it seems to work good for me now. Happy to merge whenever.
